Backend/src/main/java/com/example/backend/project/controller/ProjectController.java

```diff
@@ -34,225 +36,169 @@ public class ProjectController {
-            return new ResponseEntity("You don't have access to CDB", HttpStatus.UNAUTHORIZED);
-
-        if (!projectService.existsById(id)) return new ResponseEntity("Project not found", HttpStatus.NOT_FOUND);
-        Project project = projectService.findById(id);
-        Set<FRTeamMemberDPO> frTeamMemberDPOSet = new HashSet<>();
-        for (AppUser appUser: project.getFrteammembers()){ //iterating through project's FR team members to extract id, first and last name
-            FRTeamMemberDPO x = new FRTeamMemberDPO(
-                    appUser.getId(),
-                    appUser.getFirstName(),
-                    appUser.getLastName()
-            );
-            frTeamMemberDPOSet.add(x);
+        AppUser user = getUser(googleTokenEncoded);
+        try {
+            return new ResponseEntity(projectService.findById(id, user), HttpStatus.FOUND);
+        } catch (AuthenticationException e) {
+            return new ResponseEntity(HttpStatus.UNAUTHORIZED);
+        } catch (ProjectNotFoundException e) {
+            return new ResponseEntity(HttpStatus.NOT_FOUND);
         }
-
-        ProjectAndFRTeamMembersDPO projectAndFRTeamMembersDPO = new ProjectAndFRTeamMembersDPO(
-                project.getId(),
-                project.getIdCreator(),
-                project.getName(),
-                project.getCategory(),
-                project.getType(),
-                project.getStartDate(),
-                project.getEndDate(),
-                project.getFRResp(),
-                project.getFRGoal(),
-                project.getFirstPingDate(),
-                project.getSecondPingDate(),
-                frTeamMemberDPOSet
-        );
-
-        return new ResponseEntity(projectAndFRTeamMembersDPO, HttpStatus.OK);
     }
 
     @PostMapping()
     public ResponseEntity addProject(@RequestHeader String googleTokenEncoded, @RequestBody ProjectDTO projectDTO){
-        List<AUTHORITY> a = List.of(AUTHORITY.MODERATOR, AUTHORITY.ADMINISTRATOR);
-        String email = JwtVerifier.verifyAndReturnEmail(googleTokenEncoded);
-        if (email == null)
-            return new ResponseEntity("Token is missing or invalid", HttpStatus.UNAUTHORIZED);
-        if (userService.findByEmail(email) == null)
-            return new ResponseEntity("You don't have access to CDB", HttpStatus.UNAUTHORIZED);
-        if (!a.contains(userService.findByEmail(email).getAuthority()))
-            return new ResponseEntity("You don't have premission to this resource", HttpStatus.UNAUTHORIZED);
-
-        return new ResponseEntity(projectService.addProject(projectDTO), HttpStatus.OK);
+        AppUser user = getUser(googleTokenEncoded);
+        try {
+            return new ResponseEntity(projectService.addProject(projectDTO, user), HttpStatus.CREATED);
+        } catch (AuthenticationException e) {
+            return new ResponseEntity(HttpStatus.UNAUTHORIZED);
+        } catch (UserNotFoundException e) {
```

Comment:
Ovaj exception i status nije dobro ovdje. Znaci ako se je frontend poslao id korisnika koji ne postoji onda backend vraca 400 bad request s odgovarajucom porukom (npr. "Error in field FR responsible: User with ID 17 does not exist.")

Reply:
404 not found sam stavio jer sam htio koristiti malo specifičniji response code, ali ako želiš mogu staviti generični 400 i poruku

Reply:
Stavi 400 i poruku.
Inace, 404 se koristi samo kod poziva resursa (GET, PUT, PATCH, DELETE...) preko identifikatora (npr. api/users/13 ili /api/users/jj) ako tog resursa nema.